Add tag filtering options

ItsT-Mo opened the pull request:
Allows issue filtering based on tags.

Fixes issue #1

Empty2k12 commented on app/src/main/java/com/naman14/hacktoberfest/fragment/ExploreFragment.java:
```diff
@@ -151,6 +154,44 @@ public boolean onSelection(MaterialDialog dialog, View view, int which, CharSequ
+        new MaterialDialog.Builder(getActivity())
+                .title("Select Tags")
+                .items(Utils.getTagsArray())
+                .itemsCallbackMultiChoice(tagsPreferenceIndex
```
weird formatting

ItsT-Mo replied:
Fixed in 58883bc657b707111fa007deb1b5ad26e180dd6a

Empty2k12 commented on app/src/main/java/com/naman14/hacktoberfest/utils/Utils.java:
```diff
@@ -10,12 +10,14 @@
 import android.preference.PreferenceManager;
 import android.support.customtabs.CustomTabsIntent;
 import android.text.TextUtils;
+import android.util.Log;
```
Development Import

ItsT-Mo replied:
Ah I missed that one, thanks! :)

Empty2k12 commented on app/src/main/res/layout/layout_filter_reveal.xml:
```diff
@@ -46,22 +47,50 @@
+                <TextView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="Select Tags"
```
Please use @strings here to be future proof for #9

Empty2k12 commented on app/src/main/res/layout/layout_filter_reveal.xml:
```diff
@@ -46,22 +47,50 @@
+               <TextView
+                   android:layout_width="wrap_content"
+                   android:layout_height="wrap_content"
+                   android:text="Select Language"
```
Please use @strings here to be future proof for #9

ItsT-Mo replied:
When developing I actively decided against that. I saw there was a PR open that was adressing that, but I decided to stick with the current way Strings are embedded into the app (namely hardcoded).

I did this for 2 reasons:
a) In case he does not want to use string ressources (unlikely)
b) So when he merges this branch with the master branch it will be consistent.

I would rather use string ressources aswell. If #9 gets integrated, I will update these values with string ressource values.

Empty2k12 replied:
Seems like a fair plan, you can always rebase against master later!

Empty2k12 commented:
Building your branch crahses on my OnePlus 5, Stock 7.1.1:

```
                         E  java.lang.NullPointerException: Attempt to get length of null array
                         E      at com.naman14.hacktoberfest.utils.Utils.getTagsPreferenceString(Utils.java:164)
                         E      at com.naman14.hacktoberfest.fragment.ExploreFragment.setupFilter(ExploreFragment.java:118)
                         E      at com.naman14.hacktoberfest.fragment.ExploreFragment.onCreateView(ExploreFragment.java:87)
                         E      at android.support.v4.app.Fragment.performCreateView(Fragment.java:2343)
                         E      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1419)
                         E      at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1740)
                         E      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1809)
                         E      at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:799)
                         E      at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2580)
                         E      at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2367)
                         E      at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2322)
                         E      at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2229)
                         E      at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:700)
                         E      at android.os.Handler.handleCallback(Handler.java:751)
                         E      at android.os.Handler.dispatchMessage(Handler.java:95)
                         E      at android.os.Looper.loop(Looper.java:154)
                         E      at android.app.ActivityThread.main(ActivityThread.java:6334)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

ItsT-Mo commented:
Found the issue and fixed it.

Problem was, I was returning null when no tags preference was set. Returning an emtpy String array is better in this case, since no saved value means so preferences. Thanks for the report! :)

Fixed in 7509f85

Empty2k12 commented:
With your latest fixes this is working great. Thanks for the contribution!

I think @naman14 should merge it!

ItsT-Mo commented:
Would you mind posting a screenshot of how it looks on your phone? Because on mine, the blue circle encompasses the whole filter selectors. :)

Empty2k12 commented:
![screenshot_20171011-150548](https://user-images.githubusercontent.com/6954968/31442280-2e5b76fe-ae96-11e7-9945-63fe4bc26cc8.jpg)

This is how it looks. In my opinion it could be moved down a bit further?
